airflow/providers/databricks/triggers/databricks.py

```diff
@@ -84,13 +84,30 @@ async def run(self):
         async with self.hook:
             while True:
                 run_state = await self.hook.a_get_run_state(self.run_id)
+                notebook_error = None
                 if run_state.is_terminal:
+                    if run_state.result_state == "FAILED":
+                        run_info = await self.hook.a_get_run(self.run_id)
+                        task_run_id = None
+                        if "tasks" in run_info:
+                            for task in run_info["tasks"]:
+                                if task.get("state", {}).get("result_state", "") == "FAILED":
+                                    task_run_id = task["run_id"]
```

Comment:
This seems like so many nested blocks. if -> if -> for -> if.
Can we refactor this to reduce the nesting?

Ref: https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/too-many-nested-blocks.html

e.g. modifying to something like this could avoid some of nested branches.
```
                if not run_state.is_terminal:
                    self.log.info(
                        "run-id %s in run state %s. sleeping for %s seconds",
                        self.run_id,
                        run_state,
                        self.polling_period_seconds,
                    )
                    await asyncio.sleep(self.polling_period_seconds)
                    continue
                if run_state.result_state == "FAILED":
                    run_info = await self.hook.a_get_run(self.run_id)
                    task_run_id = None
                    for task in run_info.get("tasks"):
                        if task.get("state", {}).get("result_state", "") == "FAILED":
```

Reply:
sure on it

Reply:
done @pankajkoti , please review, did same refactoring in run now operator as well

Reply:
task["run_id"] is  different for all the tasks, should i keep a map of task run id and error message? @pankajkoti

Reply:
task run id alone won't help much to trace the task itself, no? maybe [task_key](https://docs.databricks.com/api/workspace/jobs/getrun#tasks-task_key) would help identify the task too.

I think building a list of all failed tasks and outputting it might help. The final list of failed tasks could be something like
```
failed_tasks = [
    {
        "task_key": ".....",
        "run_id": "....",
        "error":  run_output["error"] or run_state.state_message # need to construct this as per your currrent logic
    },
    {
        "task_key": ".....",
        "run_id": "....",
        "error":  run_output["error"] or run_state.state_message # need to construct this as per your currrent logic
    }
]
```


WDYT?

Reply:
And then `notebook_error = failed_tasks`

Reply:
yes, correct, i also checked task_key, on it, thanks, doing changes for both, will create a utility method, and will call for both operators

Reply:
thanks too for fixing this